istioctl/cmd/kubeinject.go

```diff
@@ -43,6 +59,127 @@ const (
 	valuesConfigMapKey = "values"
 )
 
+type ExternalInjector struct {
+	client       kube.ExtendedClient
+	clientConfig *admissionregistration.WebhookClientConfig
+}
+
+func (e ExternalInjector) Inject(pod *corev1.Pod) ([]byte, error) {
+	cc := e.clientConfig
+	if cc == nil {
+		return nil, fmt.Errorf("external injector is not available")
+	}
+	var address string
+	if cc.URL != nil {
+		address = *cc.URL
+	}
+	certPool := x509.NewCertPool()
+	certPool.AppendCertsFromPEM(cc.CABundle)
+	tlsClientConfig := &tls.Config{RootCAs: certPool}
+	if cc.Service != nil {
+		svc, err := e.client.CoreV1().Services(cc.Service.Namespace).Get(context.Background(), cc.Service.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+		namespace, selector, err := polymorphichelpers.SelectorsForObject(svc)
+		if err != nil {
+			return nil, fmt.Errorf("cannot attach to %T: %v", svc, err)
+		}
+		sortBy := func(pods []*corev1.Pod) sort.Interface { return sort.Reverse(podutils.ActivePods(pods)) }
+		pod, _, err := polymorphichelpers.GetFirstPod(e.client.CoreV1(), namespace, selector.String(), timeout, sortBy)
```

Comment:
Why are we getting first pod here / port forward ? Why not use service directly?

Reply:
webhook server in istiod service can not be reached directly from outside cluster. So we will find the istiod pod and port forward the webhook server and call it.

Reply:
Can we go thru ingress? How does this work with external istiod ?

Reply:
for external istiod case, it will  not user service in clientConfig, it will use `URL` instead.

Reply:
Thnx

Reply:
np :)